Sources/Rendering/OpenGL/VolumeMapper/index.js

```diff
@@ -947,107 +947,84 @@ function vtkOpenGLVolumeMapper(publicAPI, model) {
+      model._useSmallViewport = true;
     }
 
-    // only use FBO beyond this value
-    if (model.lastXYF <= 1.43) {
-      model.lastXYF = 1.0;
-    }
+    if (!model._animationRateSubscription) {
+      // when the animation frame rate changes recompute the scale factor
+      model._animationRateSubscription = rwi.onAnimationFrameRateUpdate(() => {
+        if (model.renderable.getAutoAdjustSampleDistances()) {
+          const frate = rwi.getRecentAnimationFrameRate();
+          const adjustment = rwi.getDesiredUpdateRate() / frate;
 
-    // console.log(`last target  ${model.lastXYF} ${model.targetXYF}`);
-    // console.log(`awin aft  ${model.avgWindowArea} ${model.avgFrameTime}`);
-    const xyf = model.lastXYF;
-
-    const size = model.openGLRenderWindow.getFramebufferSize();
-    // const newSize = [
-    //   Math.floor((size[0] / xyf) + 0.5),
-    //   Math.floor((size[1] / xyf) + 0.5)];
-
-    // const diag = vtkBoundingBox.getDiagonalLength(model.currentInput.getBounds());
-
-    // // so what is the resulting sample size roughly
-    // console.log(`sam size ${diag / newSize[0]} ${diag / newSize[1]} ${model.renderable.getImageSampleDistance()}`);
+          // only change if we are off by 15%
+          if (adjustment > 1.15 || adjustment < 0.85) {
+            model._lastScale *= adjustment;
+          }
+          // clamp scale to some reasonable values.
+          // Below 1.5 we will just be using full resolution as that is close enough
+          // Above 400 seems like a lot so we limit to that 1/20th per axis
+          if (model._lastScale > 400) {
+            model._lastScale = 400;
+          }
+          if (model._lastScale < 1.5) {
+            model._lastScale = 1.5;
+          }
+        } else {
+          model._lastScale =
+            model.renderable.getImageSampleDistance() *
+            model.renderable.getImageSampleDistance();
+        }
+        const size = model.openGLRenderWindow.getFramebufferSize();
+        model._smallViewportWidth = Math.ceil(
+          size[0] / Math.sqrt(model._lastScale)
+        );
+        model._smallViewportHeight = Math.ceil(
+          size[1] / Math.sqrt(model._lastScale)
+        );
+      });
+    }
 
-    // // if the sample distance is getting far from the image sample dist
-    // if (2.0 * diag / (newSize[0] + newSize[1]) > 4 * model.renderable.getSampleDistance()) {
-    //   model.renderable.setSampleDistance(4.0 * model.renderable.getSampleDistance());
-    // }
-    // if (2.0 * diag / (newSize[0] + newSize[1]) < 0.25 * model.renderable.getSampleDistance()) {
-    //   model.renderable.setSampleDistance(0.25 * model.renderable.getSampleDistance());
-    // }
+    // use/create/resize framebuffer if needed
+    if (model._useSmallViewport) {
+      const size = model.openGLRenderWindow.getFramebufferSize();
 
-    // create/resize framebuffer if needed
-    if (xyf > 1.43) {
+      // adjust viewportSize to always be at most the dest fo size
```

Comment:
I assume this is meant to be "dest fb size"?

Reply:
fo = framebuffer object

Reply:
But yes, I use framebuffer above when it should technically be framebuffer texture